File: 2021/python/1.py
import core
import sys

def part1(numbers):
	previous = sys.maxsize
	total = 0
	for value in numbers:
		value = int(value)
		if value > previous:
			total += 1
		previous = value
	print(total)

def part2(numbers):
	previous = sys.maxsize
	total = 0
	for i in range(0, len(numbers) - 2): 
		value = int(numbers[i]) + int(numbers[i + 1]) + int(numbers[i + 2])
		if value > previous:
			total += 1
		previous = value
	print(total)

numbers = core.load_data("1.txt")

part1(numbers)

part2(numbers)

File: 2021/1.php
<?php

require('libs/core.php');

function part1($numbers) {
	$previous = PHP_INT_MAX;
	$total = 0;
	foreach ($numbers as $value) {
		if ($value > $previous) {
			$total++;
		}
		$previous = $value;
	}
	return $total;
}

function part2($numbers) {
	$previous = PHP_INT_MAX;
	$total = 0;
	for ($i = 0; $i < count($numbers) - 2; $i++) {
		$value = $numbers[$i] + $numbers[$i + 1] + $numbers[$i + 2];
		if ($value > $previous) {
			$total++;
		}
		$previous = $value;
	}
	return $total;
}

$numbers = load_data("1.txt");

print part1($numbers);
print "\n";
print part2($numbers);
print "\n";
